### ISSUE TITLE
Ctrl+j mapping incorrectly

### ISSUE THREAD

**Issue by Gidgidonihah** [macvim-dev/macvim#908]
I cannot map Ctrl+j in a normal way. I found this because at some point my mapping for ALE stopped working.
```
nmap <silent> <C-j> <Plug>(ale_next_wrap)
" or
nmap <silent> <C-j> :ALENextWrap<CR>
```

However, the following line WILL map to `Ctrl+j`
```
nmap <silent> <C-m> <Plug>(ale_next_wrap)
```
**To Reproduce**
1. Install mccvim via homebrew
2. Install ALE (or change the line to map do a non-ale command)
3. Add the line above to the vimdc

**Expected behavior**
Ctrl-j maps to the command

**Environment (please complete the following information):**
 - Vim version
```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Feb 18 2019 23:51:40)
macOS version
Included patches: 1-950
Compiled by travis@Traviss-Mac.local
Huge version with MacVim GUI.  Features included (+) or not (-):
+acl               -farsi             +mouse_sgr         -tag_any_white
+arabic            +file_in_path      -mouse_sysmouse    -tcl
+autocmd           +find_in_path      +mouse_urxvt       +termguicolors
+autochdir         +float             +mouse_xterm       +terminal
-autoservername    +folding           +multi_byte        +terminfo
+balloon_eval      -footer            +multi_lang        +termresponse
+balloon_eval_term +fork()            -mzscheme          +textobjects
+browse            +fullscreen        +netbeans_intg     +textprop
++builtin_terms    -gettext           +num64             +timers
+byte_offset       -hangul_input      +odbeditor         +title
+channel           +iconv             +packages          +toolbar
+cindent           +insert_expand     +path_extra        +transparency
+clientserver      +job               +perl/dyn          +user_commands
+clipboard         +jumplist          +persistent_undo   +vartabs
+cmdline_compl     +keymap            +postscript        +vertsplit
+cmdline_hist      +lambda            +printer           +virtualedit
+cmdline_info      +langmap           +profile           +visual
+comments          +libcall           +python/dyn        +visualextra
+conceal           +linebreak         +python3/dyn       +viminfo
+cryptv            +lispindent        +quickfix          +vreplace
+cscope            +listcmds          +reltime           +wildignore
+cursorbind        +localmap          +rightleft         +wildmenu
+cursorshape       +lua/dyn           +ruby/dyn          +windows
+dialog_con_gui    +menu              +scrollbind        +writebackup
+diff              +mksession         +signs             -X11
+digraphs          +modify_fname      +smartindent       -xfontset
+dnd               +mouse             +startuptime       +xim
-ebcdic            +mouseshape        +statusline        -xpm
+emacs_tags        +mouse_dec         -sun_workshop      -xsmp
+eval              -mouse_gpm         +syntax            -xterm_clipboard
+ex_extra          -mouse_jsbterm     +tag_binary        -xterm_save
+extra_search      +mouse_netterm     +tag_old_static    
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/Applications/MacVim.app/Contents/Resources/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_MACVIM -Wall -Wno-unknown-pragmas -pipe  -DMACOS_X -DMACOS_X_DARWIN  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: clang   -L. -fstack-protector-strong -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib -L. -fstack-protector-strong -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib  -L/usr/local/lib -o Vim -framework Cocoa -framework Carbon       -lm  -lncurses -liconv -framework AppKit   -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE 
```
 - OS: macOS 10.14.3
 - Terminal: iTerm2 or GUI

**Comment by eirnym** [comment]
@Gidgidonihah Could you reproduce it in the build 156 which is recently out? better, both home-brew and from GitHub

**Comment by eirnym** [comment]
I can't reproduce bindings error with following bindings. Probably you've upgraded plugin or it stops working by other reason.

`:nmap <C-j> iHello<ESC>`
`:nmap <C-j> iHAHAHAHA<ESC>`

**Comment by ychin** [comment]
I couldn't quite reproduce this either. @Gidgidonihah Just to make sure I understand are you using this in GUI or terminal?

**Comment by Gidgidonihah** [comment]
I made sure to remove all plugins and test with everything vanilla. I'll pull down the latest build and verify.

I get the same result both with GUI and in terminal.

**Comment by eirnym** [comment]
@Gidgidonihah Have you tested simple bindings like above? Are you sure that your plugin works correctly?

**Comment by Gidgidonihah** [comment]
@eirnym Sorry, I was unclear. I noticed this issue *because* of a plugin binding I had been using, but simplified it and tested with the simple bindings above without any plugins installed and an empty vimconfig.

**Comment by Gidgidonihah** [comment]
Ok, after further testing it seems my initial testing was flawed. I thought I had completely isolated vanilla, but that may not have been the case. 

I'm going to go ahead and close this issue. If I'm able to recreate with a fully isolated vim, I'll reopen.

**Comment by eirnym** [comment]
You're welcome

**Comment by Gidgidonihah** [comment]
Ok, I finally carved out some time to do some more thorough testing. Turns out this is in fact, not an issue, and my tests were tainted. When I thought I had removed all installed plugins, I was wrong.

My vim was still infected with pathogen. This is what was causing the issue. Removing the pathogen plugin (which was installed differently that all the others for obvious reasons) cleared up the issue. 

After reinstalling all the original plugins (besides pathogen) using vim's native package loading everything was working as expected.

My apologies for the inaccurate bug report, and thank you for the quick responses.
